File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public void deleteNode(ListNode node) {
        //1. Set the value of the current to the next
        node.val=node.next.val;
        
        //2. Skip the pointer to the next one
        node.next= node.next.next;
        
        
    }
}

File: 0237-delete-node-in-a-linked-list/0237-delete-node-in-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteNode(self, node):
        """
        :type node: ListNode
        :rtype: void Do not return anything, modify node in-place instead.
        """
        #1. Set the current value to the next node since deletion
        node.val = node.next.val
        
        #2. The next pointer of current node should skip one
        node.next = node.next.next
        